### PR TITLE
Reference SotA in SoF, closes #6151

### LINKED ISSUE #6151: [SoF] Add Dialog Reference for SotA S20 North Knalga
### Text Issues: Typos, grammar mistakes, suggested improvements
I propose adding a reference to Secrets of the Ancients in the story scenario "In the Dwarven City". Currently, there is a dialog that goes like this:
https://github.com/wesnoth/wesnoth/blob/dd3baa488150bfc8d8cd0105da9083009cb29ca0/data/campaigns/Sceptre_of_Fire/scenarios/2t_In_the_Dwarven_City.cfg#L140-L157
With the addition of the campaign Secrets of the Ancients, the line:
>             message= _ "Ha! There hasn’t been a border attack since the men came from over the sea, why would someone attack us now?"
Is no longer 100% true, as in Secrets of the Ancients the protagonist unleashes a major attack in the heart of North Knalga. I suggest we find a way to accommodate mentioning SotA in this dialog.

### PR DESCRIPTION
**Pull request by max-torch**
Add a reference to SotA North Knalga in Sceptre of Fire.

### CODE REVIEW

**Comment by stevecotton** [comment]
I feel there ought to be a common name and lore for SotA from the non-necromancers' viewpoint, but that depends on a lore description of what non-necromancers know about SotA, and might be very different from the player knowledge of SotA.

For example, it was lead by a necromancer called Ras-Tabahn who announced himself at Carcyn. Ardonna didn't leave witnesses, so her name is probably unknown.

The desecration of Alduin's graveyard and killing of their watchmen has to be known, but is it known that a student at the university did it?

**Comment by max-torch** [comment]
> I feel there ought to be a common name and lore for SotA from the non-necromancers' viewpoint, but that depends on a lore description of what non-necromancers know about SotA, and might be very different from the player knowledge of SotA.
> 
> For example, it was lead by a necromancer called Ras-Tabahn who announced himself at Carcyn. Ardonna didn't leave witnesses, so her name is probably unknown.
> 
> The desecration of Alduin's graveyard and killing of their watchmen has to be known, but is it known that a student at the university did it?

Wouldn't Ardonna be known by the journal she left behind to be found? Or was it Malin Keshar who finds the journal? Another thing I wonder is if SotA provides any explanation towards the origins of all the other undead on the wesnoth continent like Mal ravanal, and undead during asheviere's time? They are all disciples that trace themselves to either Ardonna or Ras-Tabahn?

**Comment by stevecotton** [comment]
I'd expect anyone reading the journal to do so secretly and not spread knowledge from it, lest they themselves are branded a necromancer.

**Review by nemaara — APPROVED** [review]

File: data/campaigns/Sceptre_of_Fire/scenarios/2t_In_the_Dwarven_City.cfg [hunk 1]
```diff
@@ -155,6 +155,14 @@
             speaker=Durstorn
             message= _ "Good one, let them try to take it! Ha!"
         [/message]
+        [message]
+            speaker=Noiraran
+            message= _ "What about that time the mines of North Knalga was invaded by ancient-looking undead liches?"
```
[the remark this answers is not in the document]
Comment (max-torch) [reply data/campaigns/Sceptre_of_Fire/scenarios/2t_In_the_Dwarven_City.cfg]: Thanks.
https://github.com/wesnoth/wesnoth/pull/6152/commits/540a5c132078f4e5683c5fdb458653aba6bc1b9c

**Pull request merged**